Frontend/ui-customer/src/Helpers/Ride.purs

```diff
@@ -117,10 +117,12 @@ checkRideStatus rideAssigned = do
               else pure unit
             when (isNothing currRideListItem.rideRating) $ do
               when (resp.status /= "CANCELLED" && length listResp.list > 0) $ do
+                let nightSafetyFlow = not (fromMaybe true resp.hasNightIssue) && ((withinTimeRange "21:00:00" "06:00:00" (convertUTCtoISC(fromMaybe "" resp.rideStartTime) "HH:mm:ss")) || (withinTimeRange "21:00:00" "06:00:00" (convertUTCtoISC(fromMaybe "" resp.rideEndTime) "HH:mm:ss")))
```

Comment:
this looks repetitive code also, make it a function like suggested above then and reuse it